### PR TITLE
feat(networks): support for Arbitrum

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds support for arbitrum.

NB: On thegraph, the proper name of the network is `arbitrum-one`

https://thegraph.com/hosted-service/subgraph/unlock-protocol/arbitrum

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/networks/src/networks/arbitrum.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  baseCurrencySymbol: 'ETH',
+  locksmithUri: 'https://locksmith.unlock-protocol.com',
+  nativeCurrency: {
+    name: 'ETH',
+    symbol: 'ETH',
+    decimals: 18,
+  },
+  startBlock: 17429533,
+  previousDeploys: [],
+  isTestNetwork: true,
```
Comment (julien51) [inline packages/networks/src/networks/arbitrum.ts:36]:
```suggestion
  isTestNetwork: false,
```

**Comment by clemsos** [comment]
We lack a description for Arbitrum One. Here is a tentative one

> Arbitrum One is the flagship Arbitrum mainnet chain. It is a Layer 2 (L2) Optimistic Rollup chain running on top of Ethereum Mainnet that enables high-throughput, low cost smart contracts operations.

File: packages/networks/src/networks/arbitrum.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  locksmithUri: 'https://locksmith.unlock-protocol.com',
+  nativeCurrency: {
+    name: 'ETH',
+    symbol: 'ETH',
+    decimals: 18,
+  },
+  startBlock: 17429533,
+  previousDeploys: [],
+  isTestNetwork: false,
+}
```
Comment (clemsos) [inline packages/networks/src/networks/arbitrum.ts:37]:
```suggestion
description: "Arbitrum One is the flagship Arbitrum mainnet chain. It is a Layer 2 (L2) Optimistic Rollup chain running on top of Ethereum Mainnet that enables high-throughput, low cost smart contracts operations.",
}
```

File: packages/networks/src/networks/arbitrum.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { NetworkConfig } from '@unlock-protocol/types'
+
+export const arbitrum: NetworkConfig = {
+  publicProvider: 'https://rpc.ankr.com/arbitrum',
+  provider: 'https://rpc.ankr.com/arbitrum',
```
Comment (julien51) [inline packages/networks/src/networks/arbitrum.ts:5]:
```suggestion
  provider: 'https://rpc.unlock-protocol.com/42161',
```

**Comment by julien51** [comment]
Is this ready for prime time?

File: provider/src/index.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ const supportedNetworks = {
   '137':
     'https://snowy-weathered-waterfall.matic.quiknode.pro/5b11a0413a62a295070c0dfb25637d5f8c591aba/',
   '42220': 'https://forno.celo.org',
+  '42161': 'https://rpc.ankr.com/arbitrum',
```
Comment (clemsos) [inline provider/src/index.ts:15]: This will require redeploying the provider

**Comment by clemsos** [comment]
I ran the `configUnlock` (which was missing) and locks are created properly from the block explorer.

Then I am getting a weird provider error from hardhat when trying to create a lock. 

> $ yarn hardhat lock:create --network arbitrum
>
> An unexpected error occurred:
> 
> ProviderError: we can't execute this request
>     at HttpProvider.request (/Users/clement/Dev/unlock/unlock/node_modules/hardhat/src/internal/core/providers/http.ts:49:19)
>     at HDWalletProvider.request (/Users/clement/Dev/unlock/unlock/node_modules/hardhat/src/internal/core/providers/accounts.ts:181:36)
>     at processTicksAndRejections (internal/process/task_queues.js:95:5)
>     at EthersProviderWrapper.send (/Users/clement/Dev/unlock/unlock/smart-contracts/node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)
> 

Getting data from chain works properly though, so the provider works. 

I tried the ones from [chainlist](https://chainlist.org/)

File: provider/src/index.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ const supportedNetworks = {
     'https://cool-empty-bird.xdai.quiknode.pro/4edba942fb43c718f24480484684e907fe3fe1d3/',
   '137':
     'https://snowy-weathered-waterfall.matic.quiknode.pro/5b11a0413a62a295070c0dfb25637d5f8c591aba/',
+  '42220': 'https://forno.celo.org',
```
Comment (julien51) [inline provider/src/index.ts:14]:
```suggestion
```

**Pull request merged**